### PR TITLE
Wfc two strains

### LINKED ISSUE #25: Update text prompts for form fields
### Feature Inspiration

After review, Dr. Liefer thought some text changes would make things clearer on the form.

### Feature Description

image

### Comment

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

### LINKED ISSUE #26: Update WT data
### Feature Inspiration

The latest version of the functional connectivity data in the wild-type worm is available at: https://github.com/leiferlab/worm-functional-connectivity/blob/main/atlas/wild-type.pickle.

### Feature Description

Replace the existing WT data with the updated data https://github.com/leiferlab/worm-functional-connectivity/blob/main/atlas/wild-type.pickle

### Comment

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

### LINKED ISSUE #27: Update unc-31 data
### Feature Inspiration

The latest version of the functional connectivity data in the unc-31 knockout worm is available at: https://github.com/leiferlab/worm-functional-connectivity/blob/main/atlas/unc-31.pickle.

### Feature Description

Replace the existing unc-31 data with the updated data https://github.com/leiferlab/worm-functional-connectivity/blob/main/atlas/unc-31.pickle

### Comment

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

### PR DESCRIPTION
**Pull request by fkang-pu**
## Summary Change Description
- Uploaded pickle files for wild-type and unc-31 to MEDIA_ROOT directory, keep mock pickle files for now.
- Added a method to utils.py - a wrapper to make code cleaner and to capture all errors in different categories.
- Selected a different dataset for testing in test.py.
- Added code to test response related parameter (labels) for both wide-type and unc-31 strains.
- Upgraded django, pylint, and black to newer versions.

- Resolves #25 
- Resolves #26 
- Resolves #27
- Resovles #28

## Code Review Notes
- There is a bug in current version of wormfunconn package. Had errors with some datasets, though an error message was captured and displayed on the web page for each case. Also verified the following test cases with command line scripts.
  e.g.
  stim_neu_id="RMDDL", top_n=10, generated response plots for both wild-type and unc-31 strains.
 
  stim_neu_id="AIAL", top_n=10
  Get_responses_error: zero-size array to reduction operation maximum which has no identity

  stim_neu_id="BDUL", top-n=10
  Get_responses_error: cannot unpack non-iterable NoneType object

- Ran superlinter locally and had errors with pylint.

### CODE REVIEW

**Review by lparsons — APPROVED** [review]
Looks good to me. I had a couple of minor comments.

For the errors with the wormfunconn package, we should be sure to create issues in https://github.com/leiferlab/worm-functional-connectivity (you should have access to create issues, if you have trouble, let me know and I'll get it fixed).

I ran superlinter locally and had no issues, I use this script:
```bash
#!/usr/bin/env bash
set -euxo pipefail

# Get the current directory of this script
DIR=$(cd "$(dirname "$0")" && pwd)

docker run \
    --pull="always" \
    -e FILTER_REGEX_EXCLUDE="(\.pylintrc|migrations|static\/bootstrap.*)" \
    -e FILTER_REGEX_EXCLUDE="(\.pylintrc|migrations|static\/bootstrap.*|static\/vanillaSelectBox.*)" \
    -e LINTER_RULES_PATH="/" \
    -e IGNORE_GITIGNORED_FILES=true \
    -e RUN_LOCAL=true \
    -v "${DIR}:/tmp/lint" "github/super-linter:slim-v4"
```

You might want to prune your docker of unused containers and images with:
```bash
docker system prune
```

File: neuronsimulator/utils.py [hunk 1]
```diff
@@ -206,9 +199,10 @@ def get_reqd_params_dict(self, params_dict):
 
     def get_resp_in_ndarray(self, params_dict):
         self.params_dict = params_dict
+        app_error_dict = {}
         reqd_params_dict, app_error_dict = self.get_reqd_params_dict(params_dict)
         # default value
-        app_error_dict = {}
+        # app_error_dict = {}
```
[the remark this answers is not in the document]
Comment (fkang-pu) [reply neuronsimulator/utils.py]: fixed now.

File: neuronsimulator/utils.py [hunk 2]
```diff
@@ -437,3 +431,31 @@ def get_code_snippet_for_plot(self, params_dict):
...
+        # get code snippet for the plot
+        code_snippet, app_error_dict4 = self.get_code_snippet_for_plot(reqd_params_dict)
+        app_error_dict = {
+            **app_error_dict1,
+            **app_error_dict2,
+            **app_error_dict3,
+            **app_error_dict4,
+        }
+
+        return plot_div, resp_msg, url_query_string, code_snippet, app_error_dict
```
[the remark this answers is not in the document]
Comment (fkang-pu) [reply neuronsimulator/utils.py]: Thanks for the suggestion. Made changes to both `utils.py` and `views.py`.

**Comment by fkang-pu** [comment]
@lparsons I made the changes based on your suggestions. Also fixed a bug in my code. Django form cleaned_data changed `stim_neu_id` from None to empty string somehow. I modified my code in `utils.py` to make sure `get_response` method is not called for both cases. Ran some tests from the web again and all worked. I plan to merge into main and rollout to the server tonight if I don't hear from you.  Thanks for the superlinter script, it worked on my end except missed a couples of errors discovered by `black`.  Will investigate after my vacation.

**Pull request merged**